concordium-consensus/tests/scheduler/SchedulerTests/Helpers.hs

```diff
@@ -1,24 +1,430 @@
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
 
 module SchedulerTests.Helpers where
 
+import qualified Control.Monad.Except as Except
+import Control.Monad.RWS.Strict
+import qualified Data.Map.Strict as Map
+import qualified Data.Set as Set
+import Data.Time
+import Data.Word
+import Lens.Micro.Platform
+import Test.HUnit
+
+import qualified Concordium.Crypto.SHA256 as Hash
+import qualified Concordium.ID.Types as Types
+import qualified Concordium.Types.Accounts.Releases as Types
+import Concordium.Types.SeedState (initialSeedState)
+
 import qualified Concordium.Cost as Cost
-import Concordium.Scheduler.Types
+import qualified Concordium.GlobalState.Basic.BlockState.Account as Basic
+import qualified Concordium.GlobalState.BlockState as BS
+import qualified Concordium.GlobalState.DummyData as DummyData
+import qualified Concordium.GlobalState.Persistent.Account as BS
+import qualified Concordium.GlobalState.Persistent.BlobStore as Blob
+import qualified Concordium.GlobalState.Persistent.BlockState as BS
+import qualified Concordium.GlobalState.Persistent.BlockState.Modules as BS
+import Concordium.GlobalState.Persistent.Cache
+import qualified Concordium.GlobalState.Rewards as Rewards
+import Concordium.GlobalState.Types
+import Concordium.Logger
+import Concordium.Scheduler
+import qualified Concordium.Scheduler.DummyData as DummyData
+import qualified Concordium.Scheduler.EnvironmentImplementation as EI
+import qualified Concordium.Scheduler.Runner as SchedTest
+import Concordium.Scheduler.TreeStateEnvironment
 import qualified Concordium.Scheduler.Types as Types
-import Data.Word
+import Concordium.TimeMonad
 
-getResults :: [(a, TransactionSummary)] -> [(a, ValidResult)]
-getResults = map (\(x, r) -> (x, tsResult r))
+getResults :: [(a, Types.TransactionSummary)] -> [(a, Types.ValidResult)]
+getResults = map (\(x, r) -> (x, Types.tsResult r))
 
 -- | The cost for processing a simple transfer (account to account)
 -- with one signature in the transaction.
 --
 -- * @SPEC: <$DOCS/Transactions#transaction-cost-header-simple-transfer>
-simpleTransferCost :: Energy
+simpleTransferCost :: Types.Energy
 simpleTransferCost = Cost.baseCost (Types.transactionHeaderSize + 41) 1 + Cost.simpleTransferCost
 
-simpleTransferCostWithMemo1 :: Word64 -> Energy
-simpleTransferCostWithMemo1 memoSize = Cost.baseCost (Types.transactionHeaderSize + 41 + 2 + memoSize) 1
+simpleTransferCostWithMemo1 :: Word64 -> Types.Energy
+simpleTransferCostWithMemo1 memoSize =
+    Cost.baseCost
+        (Types.transactionHeaderSize + 41 + 2 + memoSize)
+        1
+
+simpleTransferCostWithMemo2 :: Word64 -> Types.Energy
+simpleTransferCostWithMemo2 memoSize =
+    Cost.baseCost
+        (Types.transactionHeaderSize + 41 + 2 + memoSize)
+        1
+        + Cost.simpleTransferCost
+
+-- | Monad that implements the necessary constraints to be used for running the scheduler.
```

Comment:
I think you should copy over the rest of the documentation I had. Namely why do we have a newtype, and in particular point out the "unusual" instance for logger and time monads.

Reply:
I am not sure what documentation you are referring to. The place I took this from have the same amount of documentation.